### ISSUE TITLE
RFE: please provide update for shphinx 6.x

### ISSUE THREAD

**Issue by kloczek** [sphinx-contrib/zopeext#10]
https://github.com/sphinx-contrib/zopeext/blob/adf1640d105ff63fb681b87d31062fe1c661b654/pyproject.toml#L21-L23

**Comment by kloczek** [comment]
BTW with 0.3.3 pytest shows some sphinx related warnings.
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-sphinxcontrib-zopeext-0.3.3-6.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-sphinxcontrib-zopeext-0.3.3-6.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.8.16, pytest-7.2.1, pluggy-1.0.0
libraries: Sphinx-6.1.3, zope.interface-5.5.2, sphinxcontrib.zopeext-0.3.3
base tempdir: /tmp/pytest-of-tkloczko/pytest-12
rootdir: /home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3, configfile: pyproject.toml, testpaths: tests
collected 11 items

tests/example.py .                                                                                                                                                                    [  9%]
tests/test_autointerface.py s                                                                                                                                                         [ 18%]
tests/test_ext_zopeext.py .........                                                                                                                                                   [100%]

===================================================================================== warnings summary ======================================================================================
tests/test_ext_zopeext.py::test_interfaces
tests/test_ext_zopeext.py::test_interfaces
  /usr/lib64/python3.8/importlib/__init__.py:127: RemovedInSphinx80Warning: The alias 'sphinx.util.progress_message' is deprecated, use 'sphinx.util.display.progress_message' instead. Check CHANGES for Sphinx API modifications.
    return _bootstrap._gcd_import(name[level:], package, level)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/test_autointerface.py:13: Test needs updating.
========================================================================= 10 passed, 1 skipped, 2 warnings in 0.55s =========================================================================
```

**Comment by kloczek** [comment]
And it is yet another issue. sphinx executed witj `-n` shows some warnings
```console
+ /usr/bin/sphinx-build -n -T -b man doc build/sphinx/man
Running Sphinx v6.1.3
making output directory... done
[autosummary] generating autosummary for: README.rst, index.rst
loading intersphinx inventory from http://www.sphinx-doc.org/en/master/objects.inv...
loading intersphinx inventory from https://www.zope.org/objects.inv...
intersphinx inventory has moved: https://www.zope.org/objects.inv -> https://www.zope.dev/objects.inv
intersphinx inventory has moved: http://www.sphinx-doc.org/en/master/objects.inv -> https://www.sphinx-doc.org/en/master/objects.inv
building [mo]: targets for 0 po files that are out of date
writing output...
building [man]: all manpages
updating environment: [new config] 2 added, 0 changed, 0 removed
reading sources... [100%] index
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example:1: WARNING: duplicate object description of example, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMyInterface:1: WARNING: duplicate object description of example.IMyInterface, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMyInterface.equals:1: WARNING: duplicate object description of example.IMyInterface.equals, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMyInterface.x:1: WARNING: duplicate object description of example.IMyInterface.x, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMySecondInterface:1: WARNING: duplicate object description of example.IMySecondInterface, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMySecondInterface.equals:1: WARNING: duplicate object description of example.IMySecondInterface.equals, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMySecondInterface.x:1: WARNING: duplicate object description of example.IMySecondInterface.x, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.IMySecondInterface.y:1: WARNING: duplicate object description of example.IMySecondInterface.y, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.MyImplementation:1: WARNING: duplicate object description of example.MyImplementation, other instance in README, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.MyImplementation.equals:1: WARNING: duplicate object description of example.MyImplementation.equals, other instance in README, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-sphinxcontrib-zopeext.3 { } /home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example:1: WARNING: py:mod reference target not found: zope.interface
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example:1: WARNING: py:class reference target not found: zope.interface.Interface
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.MyImplementation:1: WARNING: py:class reference target not found: optional
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/doc/example.py:docstring of example.MyImplementation:1: WARNING: py:class reference target not found: optional
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:13: WARNING: py:attr reference target not found: AutoDirective._registry
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:41:<autosummary>:1: WARNING: py:class reference target not found: zope.interface.Interface
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:41:<autosummary>:1: WARNING: py:class reference target not found: zope.interface.interface.Attribute
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:41:<autosummary>:1: WARNING: py:class reference target not found: zope.interface.interface.Method
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:41:<autosummary>:1: WARNING: py:obj reference target not found: sphinxcontrib.zopeext.autointerface.setup
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface.InterfaceAttributeDocumenter:1: WARNING: py:class reference target not found: DocumenterBridge
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:class reference target not found: zope.interface.interface.Attribute
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:class reference target not found: zope.interface.Interface
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface.InterfaceDocumenter.get_object_members:1: WARNING: py:class reference target not found: sphinx.ext.autodoc.ObjectMember
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface.InterfaceMethodDocumenter:1: WARNING: py:class reference target not found: DocumenterBridge
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:class reference target not found: zope.interface.interface.Method
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:meth reference target not found: InterfaceDocumenter.special_attrgetter
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:class reference target not found: Interface
/home/tkloczko/rpmbuild/BUILD/zopeext-0.3.3/sphinxcontrib/zopeext/autointerface.py:docstring of sphinxcontrib.zopeext.autointerface:1: WARNING: py:func reference target not found: app.add_autodoc_attrgetter
done
build succeeded, 28 warnings.
```
You can peak on fixes that kind of issues in other projects
https://github.com/RDFLib/rdflib-sqlalchemy/issues/95
https://github.com/RDFLib/rdflib/pull/2036
https://github.com/click-contrib/sphinx-click/commit/abc31069
https://github.com/frostming/unearth/issues/14
https://github.com/jaraco/cssutils/issues/21
https://github.com/latchset/jwcrypto/pull/289
https://github.com/latchset/jwcrypto/pull/289
https://github.com/pypa/distlib/commit/98b9b89f
https://github.com/pywbem/pywbem/pull/2895
https://github.com/sissaschool/elementpath/commit/bf869d9e
https://github.com/sissaschool/xmlschema/commit/42ea98f2
https://github.com/sqlalchemy/sqlalchemy/commit/5e88e6e8

**Comment by mforbes** [comment]
Hmm.  When I check the first set of warnings about the deprecated `sphinx.util.progress_message`, they seem to come from `sphinxcontrib.applehelp` (zopext does not use these)... I will have to dig to see why this is getting installed.  The second set look legit.

Thanks.  Will check into these.

**Comment by mforbes** [comment]
Release 0.4.0 has no upper bounds, but I still need to make sure there are no warnings when running with strict, so I will keep this open until I fix these.

**Comment by kloczek** [comment]
Just tested 0.4.2. On top already reported sphinjx warnings I see another one on testing using pytest
```console
ext-0.4.2-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.8.16, pytest-7.3.0, pluggy-1.0.0
libraries: Sphinx-6.1.3, zope.interface-5.5.2, sphinxcontrib.zopeext-0.4.2
base tempdir: /tmp/pytest-of-tkloczko/pytest-10
rootdir: /home/tkloczko/rpmbuild/BUILD/zopeext-0.4.2
configfile: pyproject.toml
testpaths: tests
plugins: xdist-3.2.0, hypothesis-6.71.0, anyio-3.6.2, trio-0.8.0, asyncio-0.21.0
asyncio: mode=strict
collected 11 items

tests/example.py .                                                                                                                                                                    [  9%]
tests/test_autointerface.py s                                                                                                                                                         [ 18%]
tests/test_ext_zopeext.py .........                                                                                                                                                   [100%]

===================================================================================== warnings summary ======================================================================================
tests/test_ext_zopeext.py::test_interfaces
tests/test_ext_zopeext.py::test_interfaces
  /usr/lib64/python3.8/importlib/__init__.py:127: RemovedInSphinx80Warning: The alias 'sphinx.util.progress_message' is deprecated, use 'sphinx.util.display.progress_message' instead. Check CHANGES for Sphinx API modifications.
    return _bootstrap._gcd_import(name[level:], package, level)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/test_autointerface.py:13: Test needs updating.
========================================================================= 10 passed, 1 skipped, 2 warnings in 0.54s =========================================================================
```

**Comment by mforbes** [comment]
I cannot reproduce this in my mac.  I also don't see this warning [in the actions](https://github.com/sphinx-contrib/zopeext/actions/runs/4464631013/jobs/7840950207).

```
$ pytest -ra -m 'not network'
================================================================================================= test session starts =================================================================================================
platform darwin -- Python 3.8.16, pytest-7.3.0, pluggy-1.0.0
libraries: Sphinx-6.1.3, zope.interface-5.5.2, sphinxcontrib.zopeext-0.4.2
base tempdir: /private/var/folders/m7/dnr91tjs4gn58_t3k8zp_g000000gr/T/pytest-of-mforbes/pytest-29
rootdir: /Users/mforbes/work/mmfgh/zopeext
configfile: pyproject.toml
testpaths: tests
plugins: asyncio-0.21.0, xdist-3.2.0, trio-0.8.0, cov-4.0.0, hypothesis-6.71.0, anyio-3.6.2
asyncio: mode=strict
collected 11 items                                                                                                                                                                                                    

tests/example.py .                                                                                                                                                                                              [  9%]
tests/test_autointerface.py s                                                                                                                                                                                   [ 18%]
tests/test_ext_zopeext.py .........                                                                                                                                                                             [100%]

---------- coverage: platform darwin, python 3.8.16-final-0 ----------
Coverage HTML written to dir htmlcov

Required test coverage of 85% reached. Total coverage: 95.04%

=============================================================================================== short test summary info ===============================================================================================
SKIPPED [1] tests/test_autointerface.py:13: Test needs updating.
============================================================================================ 10 passed, 1 skipped in 0.37s ============================================================================================
(.venv) Admins-MacBook-Pro-2:zopeext mforbes$ 
```

**Comment by kloczek** [comment]
Here is set of modules installed in build env in which I see that warning
<details>

```console
Package                       Version
----------------------------- -----------------
alabaster                     0.7.13
Babel                         2.12.1
blinker                       1.5
build                         0.10.0
CacheControl                  0.12.11
certifi                       2022.12.7
charset-normalizer            3.1.0
distlib                       0.3.6
distro                        1.8.0
docutils                      0.19
exceptiongroup                1.0.0
filelock                      3.9.0
findpython                    0.2.4
gpg                           1.19.0
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.4.1
iniconfig                     2.0.0
installer                     0.7.0
Jinja2                        3.1.2
libcomps                      0.1.19
lockfile                      0.12.2
markdown-it-py                2.2.0
MarkupSafe                    2.1.2
mdurl                         0.1.2
msgpack                       1.0.5
packaging                     23.0
pdm                           2.5.2
pdm-backend                   2.0.6
platformdirs                  3.1.1
pluggy                        1.0.0
Pygments                      2.15.0
pyproject_hooks               1.0.0
pyproject-metadata            0.7.1
pytest                        7.3.1
python-dateutil               2.8.2
python-dotenv                 1.0.0
pytz                          2023.2
requests                      2.28.2
requests-toolbelt             0.10.1
resolvelib                    1.0.1
rich                          13.3.4
setuptools                    67.6.1
shellingham                   1.5.0.post1
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        6.1.3
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2.dev20230415
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1.dev20230415
sphinxcontrib-qthelp          1.0.3.dev20230415
sphinxcontrib-serializinghtml 1.1.5
tomli                         2.0.1
tomli_w                       1.0.0
tomlkit                       0.11.7
typing_extensions             4.5.0
unearth                       0.9.0
urllib3                       1.26.15
virtualenv                    20.21.0
wheel                         0.40.0
zipp                          3.15.0
zope.event                    4.5.0
zope.interface                5.5.2
```
</details>

**Comment by mforbes** [comment]
Thanks. I think this might be platform dependent, but am pushing a version (4.3.0) with `nitpick=True` to test on github CI.

**Comment by kloczek** [comment]
Just please have look on versions of the modules which I have on build of that modules.
In this case crucial is `sphinx` (I'm using 6.1.3).

**Comment by mforbes** [comment]
Please upgrade `sphinxcontrib-htmlhelp==2.0.1`.  We do not depend on this, but it somehow gets imported as part of `sphinxcontrib`.

**Comment by kloczek** [comment]
My automation needs git tag 🤔 https://github.com/sphinx-doc/sphinxcontrib-htmlhelp/commit/81362a175cf405bc608fe57b303964a0a0cbc16b

**Comment by mforbes** [comment]
Should be resolved in version 0.4.3 (#16).
